src/aiida/manage/configuration/profile.py

```diff
@@ -84,12 +89,17 @@ def uuid(self, value: str) -> None:
         self._attributes[self.KEY_UUID] = value
 
     @property
-    def default_user_email(self) -> Optional[str]:
+    def config_path_resolver(self) -> AiiDAConfigPathResolver:
+        """The config_path_resolver property."""
+        return self._config_path_resolver
+
+    @property
+    def default_user_email(self) -> str | None:
         """Return the default user email."""
         return self._attributes.get(self.KEY_DEFAULT_USER_EMAIL, None)
 
     @default_user_email.setter
-    def default_user_email(self, value: Optional[str]) -> None:
+    def default_user_email(self, value: str | None) -> None:
```

Comment:
I see there are lots of changes like this in the PR, which I think pollute the _actual_ change a bit. Maybe we can update all the deprecated typing some day in one go, rather than on a per-file basis during other PRs?

Reply:
That's true, I was not expect that there are so many to fix and those are actually just warnings. 
I'll not fix make changes next time and leave it in a single PR in the future. But for this PR, the changes already did so I'll not revert those back.